agent/lib/kontena/rpc/etcd_api.rb

```diff
@@ -55,6 +55,13 @@ def delete(key, opts = {})
         {error: exc.message}
       end
 
+      # @return [Hash] {'health' => Boolean} or {'error' => String}
+      def health
+        {health: etcd.health}
+      rescue Etcd::Health::Error => error
+        {error: error.message}
```

Comment:
Looking at the etcd server implementation, yes, it's also possible to get a HTTP 503 `{"health": "false"}` response.

Never seen that when testing, not sure how the `Etcd::Health#health` code handles it ATM.